rfc/rfc-0009-supply-chain-switches-template-on-flag.md

```diff
@@ -1,3 +1,197 @@
-[Draft PR](https://github.com/vmware-tanzu/cartographer/pull/75)
+# RFC 0009 Supply Chain Switches Templates
 
-[Draft Text](https://github.com/vmware-tanzu/cartographer/blob/rfc-0009-supply-chain-switches-template-on-flag/rfc/rfc-0009-supply-chain-switches-template-on-flag.md)
+## Summary
+
+Platform creators will need to handle multiple types of workloads. While Cartographer
+supports that behavior currently through allowing multiple supply chains, it should
+reduce the number of supply chains necessary by allowing individual supply chains
+to choose between templates within a step.
+
+## Motivation
+
+A team creating a developer platform on top of Cartographer reached out because
+they maintain supply chains for their platform. They have found that there are
+many small differences in workloads that necessitate making a change to one
+template or another. Currently, such a difference can be handled by:
+
+1. Using ytt within one template to define two different objects.
+2. Creating two templates for to handle the difference and then two supply chains
+   that are the same identical except for one step that uses template A or B.
+
+The team has expressed that they prefer having separate templates (as opposed to
+maintaining ytt templates) but is concerned about the number of supply chains that
+will be needed to pursue this approach. Their hope is to keep the number of
+supply chains small, even as many small differences between workloads could lead to
+a combinatoric explosion of paths to prod.
+
+## Explanation
+
+### Current state of Cartographer
+
+Cartographer is already capable of matching a supply chain to a workload through selectors.
+With no further effort, a platform can be created that has multiple supply chains with different
+purposes or requirements. By specifying characteristics of workloads in the workload labels,
+devs can provide information to operators necessary to choose the right supply chain for
+each workload.
+
+While this behavior is currently possible, relying only on this selection would lead to a platform
+with a large number of supply chains.
+
+Workarounds, for example templates that use ytt templating to embed conditional behavior,
+would represent a code smell that this primitive is not adequate for Cartographer users.
```

Comment:
I fully expect use of ytt to manage conditional output to still be in very wide spread use. What this RFC is enabling is the ability to avoid a root level if/else where there is otherwise little commonality. In some cases a template may be able to shift away from using ytt, but I don't expect that to be common.

ytt remains the only meaningful way to:
- conditionally add/remove a field
- iterate over values
- coerce types
- apply logic
- do anything dynamic other than plain substitution

Reply:
ytt is useful. But there would be advantages to eliminating its use. [For example](https://github.com/vmware-tanzu/cartographer/pull/72#issuecomment-1028436294).

> ytt remains the only meaningful way to:

Enumerating these uses is valuable! We can begin to observe which are the most common and prioritize developing Cartographer native alternatives!